From 39b03eb8d0a385412f4869fee1119bbac060e55d Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Wed, 22 Oct 2025 17:09:27 +0200
Subject: [PATCH 5/5] x86: create new pc-q35 machine type for rhel 9.8

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 413: x86: create new pc-q35 machine type for rhel 9.8
RH-Jira: RHEL-105902
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Commit: [1/1] a18e7e378b691d92d9ec538e0efdd1a003f88972 (seott1/cos-qemu-kvm)

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/i386/pc_q35.c | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index a241f27c93..60c03c3102 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -690,23 +690,35 @@ DEFINE_Q35_MACHINE(2, 6);
 
 /* Red Hat Enterprise Linux machine types */
 
-static void pc_q35_rhel_machine_9_6_0_options(MachineClass *m)
+static void pc_q35_rhel_machine_9_8_0_options(MachineClass *m)
 {
     PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
     pc_q35_machine_options(m);
-    m->desc = "RHEL-9.6.0 PC (Q35 + ICH9, 2009)";
+    m->desc = "RHEL-9.8.0 PC (Q35 + ICH9, 2009)";
     pcmc->smbios_stream_product = "RHEL";
-    pcmc->smbios_stream_version = "9.6.0";
+    pcmc->smbios_stream_version = "9.8.0";
 
     /* NB: remember to move this line to the *latest* RHEL 9 machine */
     compat_props_add(m->compat_props, hw_compat_rhel_9, hw_compat_rhel_9_len);
+}
+DEFINE_Q35_MACHINE_AS_LATEST(9, 8, 0);
+
+static void pc_q35_rhel_machine_9_6_0_options(MachineClass *m)
+{
+    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
+    pc_q35_rhel_machine_9_8_0_options(m);
+    m->desc = "RHEL-9.6.0 PC (Q35 + ICH9, 2009)";
+    m->alias = NULL;
+    pcmc->smbios_stream_product = "RHEL";
+    pcmc->smbios_stream_version = "9.6.0";
+
     compat_props_add(m->compat_props, hw_compat_rhel_9_8,
                      hw_compat_rhel_9_8_len);
     compat_props_add(m->compat_props, pc_rhel_9_8_compat,
                      pc_rhel_9_8_compat_len);
 }
 
-DEFINE_Q35_MACHINE_AS_LATEST(9, 6, 0);
+DEFINE_Q35_MACHINE(9, 6, 0);
 
 static void pc_q35_rhel_machine_9_4_0_options(MachineClass *m)
 {
-- 
2.50.1

